Major updates for code for the controls interfaces.

pateld-git opened the pull request:
Created the UART files for the Kangaroo, and updated the SPI files for the Encoder Buffer.

SPI needs more work done to flesh out the logic for each function. More functions needed.

UART needs some more updates to flesh out each function.

Added files to src and inc for both UART and SPI

More comments needed for everything.

pateld-git committed:
Created the UART files for the Kangaroo, and updated the SPI files for the Encoder Buffer.

dstockhouse commented on control/inc/spi_interface.h:
```diff
@@ -0,0 +1,6 @@
+#ifndef __SPI_INTERFACE_H
+#define __SPI_INTERFACE_H
+
+void spiInterfaceSetup(void)
```
Missing semicolon

pateld-git replied:
Fixed

dstockhouse commented on control/src/spi_interface.c:
```diff
@@ -0,0 +1,90 @@
+
```
Add file header to your sources like I have for the other source files

pateld-git replied:
Done

dstockhouse commented on control/src/uart_Kangaroo_interface.c:
```diff
@@ -0,0 +1,76 @@
+    unsigned char temp1[] = "1,start";
+    unsigned char temp2[] = "2,start";
+    unsigned char temp3[] = "1,units 100 cm = 2048 lines";
+    unsigned char temp4[] = "2,units 100 cm = 2048 lines";
```
temp1-4 are not appropriate names for variables. This can be done with #define constants or with a single character array that you copy a new string into each time

pateld-git replied:
there more so temporary names than "temp" variables

pateld-git replied:
Changed this so that it is hard coded. I will probably not be able to change that before the demo, however, I will write comments and say some things during the demo about how it should be done

dstockhouse commented on control/src/spi_interface.c:
```diff
@@ -0,0 +1,90 @@
+ * Channel is the wiringPi name for the chip select (or chip enable) pin. Both channel 0 and channel 1 will be used.
+ * Will add logic to allow for both later.
+*************************************************************************************************************************************************/
+bool spiLogic(char* instruction)
```
This function name doesn't convey what the function does. If you intend this library to be specifically used for the encoder buffer, then calling the file spi_interface and the function spiLogic is too nonspecific. It never says even mentions the encoder buffer

pateld-git replied:
Still workshopping name.

pateld-git replied:
Found better name

dstockhouse commented on control/src/uart_Kangaroo_interface.c:
```diff
@@ -0,0 +1,76 @@
+        errorMsg = "Channel Not Homed";
+        return errorMsg;
+    }
+    else if( strcmp(readback, "e2") == 0 )
```
Should this be a lowercase e?

pateld-git replied:
Yes

dstockhouse commented on control/src/spitest.c:
```diff
@@ -0,0 +1,134 @@
+/******************************************************************************
```
This file shouldn't be here, it's just general example SPI code. Make a directory in sensors/ for workshop code like this so it doesn't clutter the main project area and prevent execution.
Also, nothing in src/ should have a main() function

pateld-git replied:
Fixed

pateld-git replied:
I have now "emptied" the files to allow for updates to test the spi base code

dstockhouse commented on control/src/spi_interface.c:
```diff
@@ -0,0 +1,90 @@
+    {
+
+    }
+    else if( trcmp(instruction, "RD") == 0 )       // instruction requires 2-5 bytes
```
There's a typo, but I recommend replacing these string variables being passed into the function with #define constants or an enumerated type

pateld-git replied:
Did replacement with #define constants

pateld-git committed:
Moved Workshopping Code
- Moved SPI interface code into its own directory in sensors
- Moved UART interface code into its own directory in sensors

pateld-git committed:
Minor Changes
- Added semicolon into spi_interaface.h
- Updated spitest.c to allow for testing of interface code

pateld-git committed:
Updates to the Encoder Buffer SPI interface
- Changed spiLogic to encoderBufferRW()
- Added Functions: registerSelect(), instructionSelect()
- Added a skeleton for bit masking
- Added new and updated function to requisite .h file

pateld-git committed:
Update for Kangaroo
-Added more comments
-Added function to requisite .h file
-Also made some edits to syntax and to the comments for the encoder buffer code files.
